keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py

```diff
@@ -0,0 +1,199 @@
+
+        object_point_clouds = tf.where(
+            sort_valid_mask[:, :, :, tf.newaxis] > 0, object_point_clouds, 0.0
+        )
+
+        return (
+            object_point_clouds,
+            object_bounding_boxes,
+        )
+
+    def _augment(self, inputs):
+        point_clouds = inputs[POINT_CLOUDS]
+        bounding_boxes = inputs[BOUNDING_BOXES]
+
+        transformation = self.get_random_transformation(
+            point_clouds=point_clouds,
+            bounding_boxes=bounding_boxes,
+        )
+        (
+            object_point_clouds,
+            object_bounding_boxes,
+        ) = self.augment_point_clouds_bounding_boxes(
+            point_clouds,
+            bounding_boxes=bounding_boxes,
+            transformation=transformation,
+        )
+
+        result = {
+            OBJECT_POINT_CLOUDS: object_point_clouds,
+            OBJECT_BOUNDING_BOXES: object_bounding_boxes,
+        }
+
+        # preserve any additional inputs unmodified by this layer.
+        for key in inputs.keys() - result.keys():
+            result[key] = inputs[key]
+        return result
+
+    # def call(self, inputs, training=True):
+    #     if training:
+    #         point_clouds = inputs[POINT_CLOUDS]
+    #         bounding_boxes = inputs[BOUNDING_BOXES]
+    #         if point_clouds.shape.rank == 3 and bounding_boxes.shape.rank == 3:
+    #             return self._augment(inputs)
+    #         elif point_clouds.shape.rank == 4 and bounding_boxes.shape.rank == 4:
+    #             batch = point_clouds.get_shape().as_list()[0]
+    #             object_point_clouds_list = []
+    #             object_bounding_boxes_list = []
+    #             for i in range(batch):
+    #                 (
+    #                     object_point_clouds,
+    #                     object_bounding_boxes,
+    #                 ) = self.augment_point_clouds_bounding_boxes(
+    #                     inputs[POINT_CLOUDS][i], inputs[BOUNDING_BOXES][i]
+    #                 )
+    #                 object_point_clouds_list += [object_point_clouds[tf.newaxis, ...]]
+    #                 object_bounding_boxes_list += [
+    #                     object_bounding_boxes[tf.newaxis, ...]
+    #                 ]
+
+    #             inputs[OBJECT_POINT_CLOUDS] = tf.concat(
+    #                 object_point_clouds_list, axis=0
+    #             )
+    #             inputs[OBJECT_BOUNDING_BOXES] = tf.concat(
+    #                 object_bounding_boxes_list, axis=0
+    #             )
+    #             return inputs
+    #         else:
+    #             raise ValueError(
+    #                 "Point clouds augmentation layers are expecting inputs point clouds and bounding boxes to "
+    #                 "be rank 3D (Frame, Point, Feature) or 4D (Batch, Frame, Point, Feature) tensors. Got shape: {} and {}".format(
+    #                     point_clouds.shape, bounding_boxes.shape
+    #                 )
+    #             )
+    #     else:
```

Comment:
does the commented out branch need to be updated for the new output format?

Reply:
Uncommented